File: returned/python/today15/reverse_bits.py
# https://leetcode.com/problems/reverse-bits/

class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0
        for i in range(32):
            place = 1 << i
            digit = n & place
            if i <= 15:
                shifted = digit << (31-i*2)
            else:
                shifted = digit >> (i-15)*2-1
            result = result | shifted
        return result

File: csharp/Leet/Old/ReverseBits.cs
namespace Old.ReverseBits
{
    public class Solution
    {
        public uint reverseBits(uint n)
        {
            var result = 0u;
            for (var i = 0; i < 32; i++)
            {
                result |= (((1u << i) & n) >> i) << (31 - i);
            }
            return result;
        }
    }
}
